web/pdf_find_controller.js

```diff
@@ -742,8 +747,17 @@ class PDFFindController {
 
     // Update the match count.
     const pageMatchesCount = this._pageMatches[pageIndex].length;
-    if (pageMatchesCount > 0) {
-      this._matchesCountTotal += pageMatchesCount;
+    this._matchesCountTotal += pageMatchesCount;
+    if (this.#updateMatchesCountOnProgress && pageMatchesCount > 0) {
+      this.#updateUIResultsCount();
+    }
+
+    if (
+      !this.#updateMatchesCountOnProgress &&
+      pageIndex === this._linkService.pagesCount - 1
+    ) {
+      // For example, in GeckoView we want to have only the final update because
+      // the Java's side provides only one object to update the counts.
       this.#updateUIResultsCount();
     }
```

Comment:
```suggestion
    if (this.#updateMatchesCountOnProgress) {
      if (pageMatchesCount > 0) {
        this.#updateUIResultsCount();
      }
    } else {
      const searchedPages = 0;
      for (const matches of this._pageMatches) {
        if (matches) {
          searchedPages++;
        }
      }
      if (searchedPages === this._linkService.pagesCount) {
        // For example, in GeckoView we want to have only the final update because
        // the Java side provides only one object to update the counts.
        this.#updateUIResultsCount();
      }
    }
```

Reply:
I introduced a field `#visitedPagesCount` in order to count again and again the pages (it could be a penalty when the number of pages is high).

Reply:
Sure, that's fine by me. Just as long as it's correctly reset everywhere, as mentioned elsewhere :-)